ddl/schema.go

```diff
@@ -109,6 +111,35 @@ func checkSchemaNotExistsFromStore(t *meta.Meta, schemaID int64, dbInfo *model.D
 	return nil
 }
 
+func onModifySchemaCharsetAndCollate(t *meta.Meta, job *model.Job) (ver int64, _ error) {
+	var toCharset, toCollate string
+	if err := job.DecodeArgs(&toCharset, &toCollate); err != nil {
+		job.State = model.JobStateCancelled
+		return ver, errors.Trace(err)
+	}
+
+	dbInfo, err := checkSchemaExistAndCancelNotExistJob(t, job)
+	if err != nil {
+		if infoschema.ErrDatabaseDropExists.Equal(err) {
```

Comment:
This error is weird, I think to call `checkSchemaExistAndCancelNotExistJob` is not appropriate. Maybe you should call `t.GetDatabase(job.SchemaID)` directly.

Reply:
addressed